lib/yao/faraday_middlewares.rb

```diff
@@ -106,9 +106,10 @@ def on_complete(env)
 
 class Faraday::Response::OSResponseRecorder < Faraday::Response::Middleware
   def on_complete(env)
-    require 'pathname'
-    root = Pathname.new(File.expand_path('../../../tmp', __FILE__))
-    Dir.mkdir(root) unless File.exist?(root)
+    require 'tmpdir'
+
+    @@tmpdir ||= Dir.mktmpdir('yao-')
+    root = Pathname.new(@@tmpdir)
```

Comment:
I like your suggestion!
Only one point to discuss: Why is this class variable here?

Reply:
And how is using yao specific env name such as `YAO_DEBUG_RECORD_OUTPUT_DIR` 
rather than general purpise env var like `TMPDIR` ?

Reply:
Thx ! 

> Only one point to discuss: Why is this class variable here?
 
To cache directory path.

Everytime `Dir.mktmpdir` is evaluated, it returns unique directory paths . 
It is a little annoying to gather JSON files 🤕  )

```
[2] pry(main)> Dir.mktmpdir('yao-')
=> "/var/folders/jj/mhdl_5tx06zfnkz5kntypj9466k1pd/T/yao-20190805-25597-nf8tot"
[3] pry(main)> Dir.mktmpdir('yao-')
=> "/var/folders/jj/mhdl_5tx06zfnkz5kntypj9466k1pd/T/yao-20190805-25597-n7733r"
[4] pry(main)> Dir.mktmpdir('yao-')
=> "/var/folders/jj/mhdl_5tx06zfnkz5kntypj9466k1pd/T/yao-20190805-25597-1o10c9t"
```

一括でガっと JSON を集めたい場合にパスが変わっちゃうのがめんどうだな〜と思ってキャッシュしてました

> And how is using yao specific env name such as YAO_DEBUG_RECORD_OUTPUT_DIR
rather than general purpose env var like TMPDIR ?

`YAO_DEBUG_RECORD_OUTPUT_DIR` is used in **yrb**, not related to **yao** ( ref https://github.com/buty4649/yao-yrb/pull/1 ) 

I'm sorry for my example was confuing 🙏

Reply:
I understand why you're using `@@tmpdir` !!!
All looks good for me!